src/mixins/filing-template-mixin.ts

```diff
@@ -61,53 +64,53 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
 
-  @Action(useStore) setAffidavit!: ActionBindingIF
-  @Action(useStore) setBusinessAddress!: ActionBindingIF
-  @Action(useStore) setBusinessContact!: ActionBindingIF
-  @Action(useStore) setCertifyState!: ActionBindingIF
-  @Action(useStore) setCooperativeType!: ActionBindingIF
-  @Action(useStore) setCorrectNameOption!: ActionBindingIF
-  @Action(useStore) setCourtOrderFileNumber!: ActionBindingIF
-  @Action(useStore) setCustodianOfRecords!: ActionBindingIF
-  @Action(useStore) setDefineCompanyStepValidity!: ActionBindingIF
-  @Action(useStore) setDissolutionDate!: ActionBindingIF
-  @Action(useStore) setDissolutionStatementStepData!: ActionBindingIF
-  @Action(useStore) setDissolutionType!: ActionBindingIF
-  @Action(useStore) setDocumentOptionalEmail!: ActionBindingIF
-  @Action(useStore) setEffectiveDate!: ActionBindingIF
-  @Action(useStore) setEntityType!: ActionBindingIF
-  @Action(useStore) setFilingId!: ActionBindingIF
-  @Action(useStore) setFolioNumber!: ActionBindingIF
-  @Action(useStore) setIncorporationAgreementStepData!: ActionBindingIF
-  @Action(useStore) setHasPlanOfArrangement!: ActionBindingIF
-  @Action(useStore) setIsAutoPopulatedBusinessNumber!: ActionBindingIF
-  @Action(useStore) setIsFutureEffective!: ActionBindingIF
-  @Action(useStore) setFoundingDate!: ActionBindingIF
-  @Action(useStore) setLegalName!: ActionBindingIF
-  @Action(useStore) setMemorandum!: ActionBindingIF
-  @Action(useStore) setNameRequestApprovedName!: ActionBindingIF
-  @Action(useStore) setNameTranslations!: ActionBindingIF
-  @Action(useStore) setOfficeAddresses!: ActionBindingIF
-  @Action(useStore) setOrgPersonList!: ActionBindingIF
-  @Action(useStore) setRegistrationBusinessAddress!: ActionBindingIF
-  @Action(useStore) setRegistrationBusinessNumber!: ActionBindingIF
-  @Action(useStore) setRegistrationBusinessType!: ActionBindingIF
-  @Action(useStore) setRegistrationBusinessTypeConfirm!: ActionBindingIF
-  @Action(useStore) setRegistrationFeeAcknowledgement!: ActionBindingIF
-  @Action(useStore) setRegistrationNaics!: ActionBindingIF
-  @Action(useStore) setRegistrationStartDate!: ActionBindingIF
-  @Action(useStore) setResolution!: ActionBindingIF
-  @Action(useStore) setRestorationApplicationDate!: ActionBindingIF
-  @Action(useStore) setRestorationApprovalType!: ActionBindingIF
-  @Action(useStore) setRestorationCourtOrder!: ActionBindingIF
-  @Action(useStore) setRestorationExpiry!: ActionBindingIF
-  @Action(useStore) setRestorationNoticeDate!: ActionBindingIF
-  @Action(useStore) setRestorationRelationships!: ActionBindingIF
-  @Action(useStore) setRestorationType!: ActionBindingIF
-  @Action(useStore) setRules!: ActionBindingIF
-  @Action(useStore) setShareClasses!: ActionBindingIF
-  @Action(useStore) setStaffPayment!: ActionBindingIF
-  @Action(useStore) setTransactionalFolioNumber!: ActionBindingIF
+  @Action(useStore) setAffidavit!: (x: UploadAffidavitIF) => void
+  @Action(useStore) setBusinessAddress!: (x: any) => void
```

Comment:
The corresponding "get" call uses a type. Can  you update the set call similarly please?
https://github.com/bcgov/business-create-ui/blob/ff820e9bd892b10fed295dd29655e7c0fb274305/src/store/store.ts#L283

Reply:
Does that mean BusinessOfficeAddress is the same as BusinessAddress? That's the only reason I left it as is. Everywhere else, I tracked down the type elsewhere in the code and updated store.ts. I thought they were named differently because they had the possibility of being different. Like for small businesses they might be the same, but some big businesses, I could envision the business operating from one address and the head office being somewhere else.

Reply:
I think so. Confirm the store state property that's being written/read.

Reply:
It's not clear from the state property if it's an OfficeAddressIF or BusinessAddressIF. They both contain a mailingAddress and a deliveryAddress, but with the OfficeAddressIF, deliveryAddress is optional. I don't understand why there are two because OfficeAddressIF is more versatile. Is this a remnant of old code?

Reply:
In store.ts:
```
    setBusinessAddress (address) {
      this.stateModel.business.officeAddress = address
    },
```

The stateModel is declared in state-model.ts as type `StateModelIF`.

In state-interface.ts:
```
// State model interface
export interface StateModelIF {
  ...
  business: BusinessIF
```

Finally, in business-interface.ts:
```
/** The Business object from the API. */
export interface BusinessIF {
  .....
  officeAddress: OfficeAddressIF
```

Reply:
Thanks @severinbeauvais. I didn't realize I could follow the breadcrumbs like that. I understand it better now!

Reply:
You're welcome :)

Thanks for the updates. Looks good.